### PR TITLE
Export epinow2_cmdstan_model() function

### PR DESCRIPTION
**Pull request by adamkucharski**
Related to #681

Exports the `epinow2_cmdstan_model()` function to allow users to specify compile-time options, such as GPU use.

- Removes the `@keywords internal` tag and adds an `@export` tag in the documentation of `epinow2_cmdstan_model()` within `R/stan.R`. This change makes the function publicly available and ensures it is listed in the `NAMESPACE` file, allowing users to directly access and utilize the function for compiling Stan models with custom settings.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/epiforecasts/EpiNow2/issues/681?shareId=8c838464-ec5d-402a-a838-10dc0e86aa3b).

### CODE REVIEW

**Comment by adamkucharski** [comment]
(Test PR from discussion with @sbfnk and others)

File: R/stan.R [hunk 1]
```diff
@@ -63,8 +63,8 @@ epinow2_rstan_model <- function(model = "estimate_infections") {
 ##'
 ##' @param model A character string indicating the model to use. One of
 ##' "estimate_infections" (default), "simulate_infections",
-##' "estimate_secondary", "simulate_secondary", "estimate_truncation" or
-##' "dist_fit".
+##' "estimate_secondary", "simulate_secondary",
+##' "estimate_truncation", or "dist_fit".
```
Comment (sbfnk) [inline R/stan.R:67]:
```suggestion
##' "estimate_secondary", "simulate_secondary", "estimate_truncation" or
##' "dist_fit".
```

Comment (sbfnk) [reply R/stan.R:67]: unrelated to Issue

**Review by sbfnk — CHANGES_REQUESTED** [review]

**Review by sbfnk — APPROVED** [review]

**Comment by jamesmbaazam** [comment]
@adamkucharski Did Workspace use the exact phrase "Related to ..."? That won't close the issue automatically when this PR is merged.

**Review by jamesmbaazam — APPROVED** [review]

**Pull request merged**